sympy/ntheory/continued_fraction.py

```diff
@@ -154,7 +155,7 @@ def untillist(cf):
 
     if period:
         y = Dummy('y')
-        solns = solve(continued_fraction_reduce(period + [y]) - y, y)
+        solns = list(solveset(continued_fraction_reduce(period + [y]) - y, y))
```

Comment:
np, there are, a couple of places (merged) `solve` tests, I will take care of those, as well.